common/src/main/java/com/gregtechceu/gtceu/core/mixins/MultiPlayerGameModeMixin.java

```diff
@@ -38,4 +47,27 @@ public class MultiPlayerGameModeMixin {
             }
         }
     }
+
+    @Inject(
+            method = {"destroyBlock"},
+            at = {@At("HEAD")},
+            cancellable = true
+    )
+    private void destroyBlock(BlockPos pos, CallbackInfoReturnable<Boolean> cir) {
+        if (
+                minecraft.player == null ||
+                minecraft.level == null ||
+                !GTToolType.MINING_HAMMER.is(minecraft.player.getMainHandItem()) ||
```

Comment:
maybe let's not hardcode this to the hammer, but instead add an item tag for AOE tools?

Reply:
I am not very familiar with data-generating tags. I see that there is a CustomTags class to hold recipe tags, but other than that I don't see a good place to store a TagKey.

Reply:
CustomTags is exactly the place for those
`ItemTagLoader` is where you can add data-generated values for those tags

Reply:
CustomTags is all our non-material/prefix tags currently, it holds everything